test/statetest/statetest_loader.cpp

```diff
@@ -207,29 +204,31 @@ state::Transaction from_json<state::Transaction>(const json::json& j)
         o.max_priority_gas_price = from_json<intx::uint256>(j.at("maxPriorityFeePerGas"));
     }
 
-    if (j.contains("accessList"))
-        o.access_list = from_json<state::AccessList>(j.at("accessList"));
+    // The following fields are optional or arrays in a TestMultiTransaction:
+
+    if (const auto input_it = j.find("input"); input_it != j.end())
+        o.data = from_json<bytes>(*input_it);
+
+    if (const auto gas_it = j.find("gas"); gas_it != j.end())
+        o.gas_limit = from_json<int64_t>(*gas_it);
+
+    if (const auto value_it = j.find("value"); value_it != j.end() && !value_it->is_array())
+        o.value = from_json<intx::uint256>(*value_it);
+
+    if (const auto ac_it = j.find("accessList"); ac_it != j.end())
+        o.access_list = from_json<state::AccessList>(*ac_it);
 
     return o;
 }
 
 static void from_json(const json::json& j, TestMultiTransaction& o)
 {
-    if (j.contains("gasPrice"))
-    {
-        o.kind = state::Transaction::Kind::legacy;
-        o.max_gas_price = from_json<intx::uint256>(j.at("gasPrice"));
-        o.max_priority_gas_price = o.max_gas_price;
-    }
-    else
-    {
-        o.kind = state::Transaction::Kind::eip1559;
-        o.max_gas_price = from_json<intx::uint256>(j.at("maxFeePerGas"));
-        o.max_priority_gas_price = from_json<intx::uint256>(j.at("maxPriorityFeePerGas"));
-    }
-    o.sender = from_json<evmc::address>(j.at("sender"));
-    if (!j.at("to").get<std::string>().empty())
-        o.to = from_json<evmc::address>(j["to"]);
+    const auto base_tx = from_json<state::Transaction>(j);
```

Comment:
I think this is weird, it ignores the `base_tx.access_list` field for example.

Reply:
In multi-transaction we have a list of access lists. See below.

Reply:
Yeah I figured, still think all these formats are confusing and hiding details.

Reply:
I'm not sure what are you complaining about? The JSON format?

Reply:
The only complaint is that we never verify is stray fields are present (in this case `accessList` when we expect `accessLists`), but maybe that is part of how these formats work -- they allow any stray field.

Reply:
I have an idea how to improve it.

Reply:
Updated.